Check drivers (lab3)

Gaydarenko opened the pull request:
…tringdriver"

Gaydarenko committed:
release "class JSONFileDrive", "class PICKLEFileDriver", "class JSONStringdriver"

Gaydarenko committed:
release "def save" in linkedlist.py

Gaydarenko committed:
release "def load" in linkedlist.py

Gaydarenko committed:
added:
class SDBuilder
class JSONFileBuilder(SDBuilder)
class PICKLEFileBuilder(SDBuilder)
class JSONStringBuilder(SDBuilder)

Gaydarenko committed:
added:
class SDFabric

EnotYoyo commented on Drivers.py:
```diff
@@ -0,0 +1,123 @@
+    @staticmethod
+    def get_sd_driver(driver_name: str, t=True):
+        while t:
+            if driver_name == "JSONFileDriver":
```
В таких ситуация часто используют dict: 
```python
builders = {
    "JSONFileDriver": JSONFileBuilder,
    "PICKLEFileDriver": PICKLEFileBuilder,
    # ...
}

driver = builders.get(driver_name)
if driver is not None:
    return driver()

```

Gaydarenko replied:
fixed

Gaydarenko committed:
corrected "class SDFabric"

Gaydarenko committed:
delete rudiment "t" in "class SDFabric"

Gaydarenko committed:
def set_structure_driver

Gaydarenko committed:
upgrade "def save" - in "next" & "prev" writing id(node)
upgrade "def load" - append Node from value

Gaydarenko committed:
add self.sd in __init__

upgrade "def save" - теперь метод ничего не возвращает, а только вызывает write от переданного драйвера

EnotYoyo reviewed (approved):
В методе load еще нужно добавить sd.read() и все. 
Я считаю, что Вы это без труда сделаете, поэтому задание считаю засчитанным.

Gaydarenko committed:
upgrade "def load" - теперь метод ничего не принимает на вход.
Исправлен баг в "def load" - в ловарь обращаюсь теперь через str(id_head)
